src/modules/dataman/dataman.cpp

```diff
@@ -540,7 +540,15 @@ _file_write(dm_item_t item, unsigned index, dm_persitence_t persistence, const v
 static void
 _ram_flash_update_flush_timeout()
 {
-	dm_operations_data.ram_flash.flush_timeout_usec = hrt_absolute_time() + RAM_FLASH_FLUSH_TIMEOUT_USEC;
+	timespec &abstime = dm_operations_data.ram_flash.flush_timeout;
+
+	if (clock_gettime(CLOCK_REALTIME, &abstime) == 0) {
+		const unsigned billion = 1000 * 1000 * 1000;
+		uint64_t nsecs = abstime.tv_nsec + (uint64_t)RAM_FLASH_FLUSH_TIMEOUT_USEC * 1000;
+		abstime.tv_sec += nsecs / billion;
+		nsecs -= (nsecs / billion) * billion;
+		abstime.tv_nsec = nsecs;
```

Comment:
We have this function everywhere, can we re-use it?

https://github.com/PX4/Firmware/blob/45a53726d6594f6285243c1f1198fa431958e1c5/platforms/posix/src/px4/common/drv_hrt.cpp#L530-L535

Reply:
We don't need to convert, we need one that adds an offset here, and I have not seen that.

Reply:
Ah ok.